src/Surfnet/StepupMiddleware/ApiBundle/Identity/Repository/RaCandidateRepository.php

```diff
@@ -201,6 +201,27 @@ public function createSearchQuery(RaCandidateQuery $query)
         return $queryBuilder->getQuery();
     }
 
+    /**
+     * @param RaCandidateQuery $query
+     * @return \Doctrine\ORM\Query
+     */
+    public function createOptionsQuery(RaCandidateQuery $query)
```

Comment:
For future maintainability, it might be a good idea to move this logic (that is duplicated in the other repositories) to a helper class that is injected into those repos. This prevents code duplication and improves maintainability. But only if this does not raise other problems.

Reply:
I think for now I find the readability better than to split the minimal code duplication. The required (number of) columns do also have some differences.

Reply:
Agreed!